.circleci/config.yml

```diff
@@ -355,6 +382,201 @@ jobs:
       - store_artifacts:
           name: "Store logs"
           path: /tmp/e2e
+  e2e_testing:
+    parameters:
+      k8sVersion:
+        description: version of k3s to use or "kind" and "kindIpv6"
+        type: string
+        default: << pipeline.parameters.last_k8s_version >>
+      parallelism:
+        description: level of parallelization
+        type: integer
+        default: 1
+      target:
+        description: makefile target without test/e2e prefix
+        type: string
+        default: ""
+      arch:
+        description: The golang arch
+        type: string
+        default: amd64
+      cniNetworkPlugin:
+        description: The CNI networking plugin to use [flannel | calico]
+        type: string
+        default: flannel
+      legacyKDS:
+        description: if should run tests with new implementation of KDS
+        type: boolean
+    executor:
+      name: vm-<< parameters.arch >>
+    parallelism: << parameters.parallelism >>
+    steps:
+      - checkout
+      - run:
+          name: Download build output from GitHub action
+          command: |
+            echo "Running with: \
```

Comment:
Hmm could this be a shell script? Could be easier to test

Reply:
I did use a separated file to test when authoring locally.
This script may be refactored after the transition from CircleCI to GH Actions. In addition, this script is only useful on the CI, we don't use it anywhere else.